File: leetcode/Intersection of Two Linked Lists/1.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getIntersectionNode(headA, headB *ListNode) *ListNode {
	mark := map[*ListNode]struct{}{}

	for headA != nil {
		mark[headA] = struct{}{}
		headA = headA.Next
	}

	for headB != nil {
		_, ok := mark[headB]
		if ok {
			return headB
		}
		headB = headB.Next
	}

	return nil
}

File: leetcode/Remove Nth Node From End of List/1.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun removeNthFromEnd(head: ListNode?, n: Int): ListNode? {
        var curr = head
        var length = 0
        while (curr != null) {
            length++
            curr = curr.next
        }
        curr = head

        val move = length - n
        if (move == 0) return curr?.next

        for (i in 0 until move - 1) {
            if (curr == null) break
            curr = curr.next
        }

        curr?.next = curr?.next?.next
        return head
    }
}